third_party/cpplint.py

```diff
@@ -4480,7 +4480,8 @@ def _ClassifyInclude(fileinfo, include, is_system):
             or include.endswith(".hpp") \
             or include.endswith(".hxx") \
             or include.endswith(".H") \
-            or include.endswith(".hh")
+            or include.endswith(".hh") \
+            or '.' not in include[include.rfind('/')]  # no extension
```

Comment:
Are we sure we want this? I recall you saying, that that's in order to have `#include <cuda/memory_resource>` recognized as C++. However, we are not really running linter for third party headers, so I don't feel it's necessary to run it for `<cuda/memory_resource>`. On the other hand, this may lead to strange linter behaviour in future.

@banasraf , what's your opinion about it?

Reply:
If I understand correctly, it's not about running linter on a third party header, but about ordering of includes. Linter requires some certain order of includes, and this change is to correctly qualify the cuda/memory_resource as C++ include.

Reply:
^^^ exactly that.